tests/test_client_functional.py

```diff
@@ -3610,3 +3610,30 @@ async def not_ok_handler(request):
         "/ok", timeout=aiohttp.ClientTimeout(total=0.01)
     ) as resp_ok:
         assert 200 == resp_ok.status
+
+
+async def test_request_with_wrong_ssl_type(aiohttp_client: AiohttpClient) -> None:
+    app = web.Application()
+    session = await aiohttp_client(app)
+
+    with pytest.raises(TypeError, match="ssl should be SSLContext, Fingerprint, .*"):
+        await session.get("/", ssl=42)  # type: ignore[arg-type]
+
+
+async def test_request_with_wrong_proxy(aiohttp_client: AiohttpClient) -> None:
+    app = web.Application()
+    session = await aiohttp_client(app)
+
+    with pytest.raises(TypeError):
```

Comment:
Can't understand why my tests can't catch `InvalidURL` here :(

Reply:
Probably because `TypeError` is not a subclass of `ValueError`.

Reply:
Great sign for me to go to sleep)) thanks